avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -491,6 +491,9 @@ private static Object fromConnectData(
               requireContainer);
 
         case BYTES: {
+          if (value instanceof BigDecimal) {
```

Comment:
because now we are enforcing the semantics: decimal type is preserved iff the precision is specified, otherwise it will be convert to bytes. Therefore, it can happen that some Connect data, the type is Decimal without precision; when we convert its schema to avro schema, the type becomes bytes. Therefore, in `fromConnectData()`, even though the schema type is bytes, it is possible for the data to be converted to be a BigDecimal. I added this conversion because I actually encountered a bug caused by the fact.